src/DPoPService.ts

```diff
@@ -0,0 +1,104 @@
+import { CryptoUtils, JwtUtils } from "./utils";
+import { IndexedDbCryptoKeyPairStore } from "./IndexedDbCryptoKeyPairStore";
+
+/**
+ * Provides an implementation of Demonstrating Proof of Possession (DPoP) as defined in the
+ * OAuth2 spec https://datatracker.ietf.org/doc/html/rfc9449.
+ */
+
+export interface GenerateDPoPProofOpts {
+    url: string;
+    accessToken?: string;
+    httpMethod?: string;
+}
+export class DPoPService {
+    public static async generateDPoPProof({
```

Comment:
please make this a normal service like all thers:
- constructor
- no static methods
- instantiate it within the `UserManager` constructor
- like `this._silentRenewService = new SilentRenewService(this);`